### ISSUE TITLE
Project Scope Idea

### ISSUE THREAD

**Issue by LcNguyn** [nguyenducminhtam/sepm#23]
**USER ACTIVITY FLOW**
 
**Target User**
Kids: Simplicity, Attraction in the process are top priorities
Non- English: Reduce using English => Using Vietnamese for UX (User Experience) or Creative Images

**Overall**
The app will have 2 main options: 
Camera - Explore
Review - Book  

**Action**
_1. Create Account_
User can create (type) their own unique name, age
Since it is just a kid app so creating an account with a password might not be necessary. 
Idea: There will be a cute companion animal (avatar) throughout the user's journey in the app to create happiness, motivation for the kids.
         	+ Companion animal: an image(s). It just pops in/out in some situations.

_2. Tutorial_
OTP Tutorial to guide users using this application.

_3. Explore_
The screen will be a full-screen camera, there will be a dictionary button for users to go to the Book tab.
When users move the camera around them to choose the item they need to translate, the corresponding potential words will appear on the screen along with a representative image. 
When users click on any word to learn, it will be saved in the Book.
 
_4. Book/Review:_
All of the saved words are in this place for the user to review them whenever they want. 
There is a button that links to the Quiz whenever the user wants to take tests.
Users choose a quiz button to have a test with a limit of 10 words. 

_5. Quiz_
The quiz will be divided into different categories according to the number of words they want to challenge himself (10 words test, 20 words test,...). The words will not be repeated for each category
For example:
The user has explored 100 words
After that, he does the Random 10 words level.
When he does the test again, there will be another 10 words that he has learned not the old 10 words

+_Why this category but not the Difficulty level or base on Time?_
We cannot evaluate if the words are difficult to learn or not.
If based on time, we might have to limit the number of words the user can explore each day.
We cannot make sure that they have tested all the words if the words are classified by Difficulty level

_6. Reward_
Based on how long the user takes the test, how many correct words in the test,... that an accordingly mark will be given. Enough mark will increase user level and at a specific level, he will be rewarded an online sticker (animal) to his collection.

**Comment by LcNguyn** [comment]
This is my idea that I have been researching and come up with. So let's discuss its pros and cons

**Comment by trungduong0103** [comment]
> This is my idea that I have been researching and come up with. So let's discuss its pros and cons

let's discuss this during our metting

**Comment by s3687636** [comment]
@LcNguyn can you share a link to the note that we came up after our meeting. it can be a google doc, so that we all can review and adjust if required.

**Comment by LcNguyn** [comment]
> @LcNguyn can you share a link to the note that we came up after our meeting. it can be a google doc so that we all can review and adjust if required.

I have just updated the scope above. Doc link will be added soon but it is all about the scope above :)) except the report task assignment.

**Comment by LcNguyn** [comment]
### USER ACTIVITY FLOW

**Summary**
An _educational mobile application_ supports kids ( generation Alpha & Z) to do _quizzes that are specifically designed to keep track of users’ studying progress & generate their study maps_ in order to have their best study outcome. 
Furthermore, users’ interaction with the surrounding environment is also enhanced by _using modern technology such as voice and image recognition functions to educate kids_ simple English words from objects around their own house.

**Target User**
_Children_:	Kindergarten to Primary school => Moderate Simplicity. Attraction in the process 
are top priorities
_Non - English_:  Avoid using complex English => Use Creative Images (Less refer: using simple 
mother language since the app is presented to English speaking stakeholders)

**Overall**
The app will have 3 main options:
Updated: We will focus more on Quiz, other functions are the same.

_1. Quiz_ 
A nearly perfect quiz system based on an effective educating process that allows optimizing user’s learning speed. (most important, based on Anna expectation)

_2. Explore_ 
Camera, Image recognition. This function flow is still the same or reduced (if reduced, we need a meeting to choose which functions.)

_3. Notebook_
For Revision purpose, Words learned from Explore and Quiz. It can be thought as History, Review, Book,... but I use Notebook since it is familiar and easiest for children to understand

**Action**
_1. Create Account_
User can create (type) their own unique name, age.
Since it is just a kid app so creating an account with a password might not be necessary. Password maybe 10 simple, memorable images representing 10 numbers 0-9 to make passwords.
There will be a cute companion animal (avatar) throughout the user's journey in the app to create happiness, motivation for the kids.

_2. Tutorial_
OTP Tutorial to guide users using this application.

_3. Quiz_
Based on the model app, to learn a language (my exp is Japanese), there are 7 stages. Each stage has 8 - 20 levels classified by alphabet, words’ complexity and each level have 5 minor-levels and each minor-level has 3-5 tests. :)))) (There is a map for you guys to understand more.)
However, of course, we can not do it all so I suggest _removing the “stage”_ so that the biggest milestone is levels. Also, _reducing the number of levels, tests,..._ could make the app less complex for us.

_Types of quizzes, based on Duo***** :))))) :_

- Multiple choices (with/without images)
- Type
- Select: Click or Drag & Drop
- Pairs

We can change the UI to make these types different from other apps.
So far, we have come up with some new Idea: 

- Ask for Basic ingredients to make basic meals, drink (coffee, lemonade, pancake,...)
- Take photos of multiple things. Ex: 5 things have color Red or 5 different colors or follow a list of items.

About the question in the tests, at each level, there are new words and the questions are all about them (mix with few old words). No matter which minor-lvl it is, it just repeats the questions (the amount of questions increases as the minor-level increases.)
If our quiz system is inspired by them, we need ~ 20 questions and 4 new words + 4 objects for each level.

_Estimation, to conquer 1 language:_
20 questions * 17 levels *  7 stages = 2380 questions
4  words * 17 lvls * 7 stages = 476 words and 476 objects

_4. Explore_
The screen will be a full-screen camera, there will be a button for users to go to the NoteBook tab.
When users move the camera around them and snap to choose the item they need to translate, the corresponding potential words will appear on the screen along (with a representative image). Users can click on any word to hear its pronunciation or save it in the Book.

_5. Notebook_
All of the saved words are in this place for the user to review them whenever they want.
There is a button that links to the Quiz whenever the user wants to take tests.

_6. Reward_
Based on how long the user takes the test, how many correct words in the test,... that an accordingly mark will be given. Enough marks will increase user level and at a specific level, he will be rewarded with an online sticker (animal, badge) to his collection.
Also, we can do the marking easier by adding 10 XP for each finished level and more XP for harder ones.

**Comment by LcNguyn** [comment]
![Modal Quizz](https://user-images.githubusercontent.com/37359318/77752455-28acc400-705a-11ea-9bb2-830743568121.PNG)

**Comment by trungduong0103** [comment]
> However, of course, we can not do it all so I suggest removing the “stage” so that the biggest milestone is levels. Also, reducing the number of levels, tests,... could make the app less complex for us.

I agree, we should reduce stages AND levels to make our app faster to develop. We should have a meeting or a discussion on how many stages should we have and what are the objectives and labels of those stages (e.g: fruits, colors, objects, and harder levels like he/she/it and more).

>If our quiz system is inspired by them, we need ~ 20 questions and 4 new words + 4 objects for each level.
 
I'm ok with 20 questions (can be reduced), but we should follow their approach to have similar questions but with different approach. 
I think we have 3 layers and that should be enough ? A Language has many Stages, each stages has many questions.

What do you guys think ? @s3687636 @thienhieu215

**Comment by thienhieu215** [comment]
> What do you guys think ? @s3687636 @thienhieu215

I agree with Trung since this is just an app to learn basic English so we don't need levels. Same with test (or we can do that after we finish the main functions)
We can increase the questions of each stage, that would be better and less confusing for kids. We can give kids rewards at some specific milestones of each stage.

**Comment by s3687636** [comment]
> _Types of quizzes, based on Duo***** :))))) :_
> 
> * Multiple choices (with/without images)
> * Type
> * Select: Click or Drag & Drop
> * Pairs

can add more: 
* read the words or sentence
* read the meaning of the words

**Comment by s3687636** [comment]
> this is just an app to learn basic English so we don't need levels

I don't think so. for my understanding, levels are just a kind of milestone with rewards as motivating signs. The kids should recognize their learning progress, they need signs, just like other games pattern. The thing here is that the level is automated by the app, hence it should not cause any complications for the users (kids).
yes, simple layers (levels, sublevels, and quizzes) are enough for our app at this stage of development, as well as our limited time frame.

**Comment by thienhieu215** [comment]
> for my understanding, levels are just a kind of milestone with rewards as motivating signs

Totally agree with you. Since Loc's new scope is like this so I just want to make it simpler:


> _Estimation, to conquer 1 language:_
> 20 questions * 17 levels * 7 stages = 2380 questions
> 4 words * 17 lvls * 7 stages = 476 words and 476 objects
